back/src/domains/convention/use-cases/notifications/NotifyIcUserAgencyRightChanged.ts

```diff
@@ -34,7 +38,13 @@ export class NotifyIcUserAgencyRightChanged extends TransactionalUseCase<
     );
     if (!user) throw errors.user.notFound({ userId: params.userId });
 
-    if (!params.roles.includes("to-review"))
+    const agencyRight = user.agencyRights.find(
+      (agencyRight) => agencyRight.agency.id === params.agencyId,
+    );
+
+    if (!agencyRight) throw errors.user.noRightsOnAgency(params);
+
+    if (!agencyRight.roles.includes("to-review"))
```

Comment:
j'ai l'impression que l'intention est assez différente de ce qu'il y avait avant, ça me semble pas être la même info qu'on va simplement chercher ailleurs : là on regarde si dans les droits stockés de l'utilisateur pour cette agence, le role to-review n'est pas contenu

Reply:
Avant on passait objet de type UserParamsForAgency en paramètre.
Mais la proposition ici est de récupérer directement ces mêmes infos depuis le repository, afin de n'avoir en paramètre du usecase que un objet de type WithAgencyIdAndUserId.